In [1]:
import pandas as pd
import pickle

In [2]:
with open('..\Base de Dados\census.pkl', mode = 'rb') as f:
    X_census_train, y_census_train, X_census_test, y_census_test = pickle.load(f)

In [3]:
X_census_train

array([[-0.2444502 , -0.17429511, -0.26209736, ..., -0.14592048,
        -0.21665953, -0.03542945],
       [-0.2444502 , -0.17429511, -0.26209736, ..., -0.14592048,
        -0.21665953,  3.20416118],
       [-0.2444502 , -0.17429511,  3.81537614, ..., -0.14592048,
        -0.21665953, -0.03542945],
       ...,
       [-0.2444502 , -0.17429511, -0.26209736, ..., -0.14592048,
        -0.21665953, -0.03542945],
       [-0.2444502 , -0.17429511,  3.81537614, ...,  0.89508344,
        -0.21665953,  0.36951938],
       [-0.2444502 , -0.17429511, -0.26209736, ...,  0.20572978,
        -0.21665953,  0.61248868]])

In [17]:
from sklearn.naive_bayes import GaussianNB

Instancia a classe do Naive Bayes

In [18]:
naive_census = GaussianNB()
naive_census.fit(X_census_train, y_census_train)

GaussianNB()

In [19]:
previsoes = naive_census.predict(X_censust_test)
previsoes

array([' >50K', ' >50K', ' >50K', ..., ' <=50K', ' >50K', ' >50K'],
      dtype='<U6')

In [20]:
y_census_test

array([' <=50K', ' <=50K', ' <=50K', ..., ' <=50K', ' <=50K', ' <=50K'],
      dtype='<U6')

In [29]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [24]:
accuracy_score(y_census_test, previsoes)

0.4767656090071648

In [27]:
confusion_matrix(y_census_test, previsoes)

array([[1172, 2521],
       [  35, 1157]], dtype=int64)

In [31]:
print(classification_report(y_census_test, previsoes))

              precision    recall  f1-score   support

       <=50K       0.97      0.32      0.48      3693
        >50K       0.31      0.97      0.48      1192

    accuracy                           0.48      4885
   macro avg       0.64      0.64      0.48      4885
weighted avg       0.81      0.48      0.48      4885



recall = % que o algoritimo consegue identificar 0.97% das pessoas que ganham >50k

precision = Quando identificado, a precisão de acerto é de 0.31%
